tests/integration_tests/llms/test_openai.py

```diff
@@ -51,6 +51,13 @@ def test_openai_extra_kwargs() -> None:
         OpenAI(model_kwargs={"model": "text-davinci-003"})
 
 
+def test_openai_call_creds_at_invocation() -> None:
```

Comment:
Thoughts on adding a test to make sure two LLMs can set different credentials, and they don't override each other? 

e.g LLM 1 have a valid credential, LLM 2 doesn't, and LLM 1 can get the string but LLM 2 can't

Reply:
i think that's probably already covered by the unit test in file below (if the openai global api key isn't being set then there's no way for one objects instance vars to overwrite another, right?)

Reply:
Yah that works too. I wish there is a more direct integration test, where different OpenAI() llms can be initialized with different api keys and they have isolated behaviors.